### PR TITLE
Flow status+fixes

### PR DESCRIPTION
**Pull request by RikardLegge**
I've tried dividing up the changes into smaller chunks to make it easier to review the changes.

Summary:
1. Edited the regex to get the column in the format 00,00 instead of only the first part 00.
2. Added two new directives 'lint-view' and 'use-server'. I also saw that 'use-server' also was requested in another pull request, so wrote this code in the same way.
4. Added more advanced linter run rules to allow grabbing the code directly out of the view buffer.
5. Changed the view formatting and added more detailed selection through the near variable.

Feedback would be appreciated.

Note: This is the first time i come in contact with python code, so please correct me if i've written it in a way which doesn't follow the best practices for the language.

### CODE REVIEW

File: linter.py [hunk 1]
```diff
@@ -72,23 +105,35 @@ def split_match(self, match):
         if match:
             open_file_name = os.path.basename(self.view.file_name())
             linted_file_name = match.group('file_name')
-
             if linted_file_name == open_file_name:
+
                 message_title = match.group('message_title')
                 message = match.group('message')
-                message_footer = match.group('message_footer') or ""
+                message_footer = match.group('message_footer') or None
```
Comment (frantic) [inline linter.py:112]: you don't need `or None`, `match.group` returns `None` if nothing found

File: linter.py [hunk 2]
```diff
@@ -72,23 +105,35 @@ def split_match(self, match):
...
+                        message = '[{0}], {1}'.format(
+                            message_title,
+                            message
+                        )
+
+                # Get the current line
                 line = max(int(match.group('line')) - 1, 0)
-                col = int(match.group('col')) - 1
+                # Get the start and ending indexes of the column
+                col_start,col_end = (int(part) for part in match.group('col').split(','))
```
Comment (frantic) [inline linter.py:131]: nit: space after comma

**Comment by frantic** [comment]
Note that is is not the upstream repo and I'm not maintaining the plugin :)
See https://github.com/SublimeLinter/SublimeLinter-flow

The PR looks good to me except I don't really like the idea of using `flow check-contents` because it doesn't honor absence of `@flow` annotation (maybe we should file a bug about this?) In any case making sure the plugin behaves the same way flow does is going to be tricky (for example, will this honor `ignore` path in `.flowconfig`?). But I guess it's not a big deal, everything is moving really fast.

**Comment by RikardLegge** [comment]
I've not used git all that much for project collaboration earlier, so thank you for noting the mistake. 
I've also fixed the parts you commented on. 

I added the "check-contents" since it gave a more up to date version of the file. Without it, the file gets linted as of the last time it was saved. (This isn't a problem with the load/save trigger, but using the background trigger causes smaller issues.)

To honor the @flow attribute is quite easy, just a check if the attribute is present, if not then don't lint. Or are there more ways the @flow attribute can be used?
The .flowconfig part is a bit trickier. One way would be to read that file and cross reference it with the current file which is to be linted. If sublime caches the file, this would probably work in most cases.

Anyhow, i'll re-fork the project, and this time i'll try to get the upstream repo ;)

**Pull request closed without merging**